compiler/rustc_codegen_llvm/src/context.rs

```diff
@@ -881,6 +881,8 @@ impl<'ll> CodegenCx<'ll, '_> {
         ifn!("llvm.lifetime.start.p0i8", fn(t_i64, ptr) -> void);
         ifn!("llvm.lifetime.end.p0i8", fn(t_i64, ptr) -> void);
 
+        ifn!("llvm.is.constant", fn(...) -> i1);
```

Comment:
This being variadic is causing tons of problems, however, something like `void` causes it to segfault (unexpectedly). Is there any way to represent an overloaded intrinsic, or will we unfortunately have to define it on each primitive individually?

Reply:
Why do you want to make it variadic? The Rust function signature you started with, taking a generic `T`, was just fine.

Reply:
Oh, you mean only on the LLVM side. No idea, sorry.

Reply:
Yea, this intrinsic is not variadic. Just like the other intrinsics in this function that use `.i32`, `.i64`, ... `is.constant` needs to be replicated for each type: https://llvm.org/docs/LangRef.html#llvm-is-constant-intrinsic

One possible solution could be to make it take `ptr` and then make all call sites pass a pointer to the value they got (cast to `void*` or sth). But I don't know if LLVM preserves constness across those casts (I'd hope it would).

If that doesn't work, you need to declare it for all the integer types and make the Rust -> LLVM intrinsic logic dispatch to the right one depending on the type.

Reply:
Oh, the latter scheme may actually be simple: convert the Rust type to an LLVM type and get the string representation of the LLVM type. Then append that to the intrinsic name.

Reply:
I am confused. When you say variadic, do you mean generic?

Reply:
in C variadic can be used as "runtime-generic". LLVM intrinsics can be somewhat generic (the type is part of the name), but we have no way of going from a Rust-generic intrinsic to an LLVM-generic intrinsic without doing the string append dance I described above.

Reply:
> Oh, the latter scheme may actually be simple: convert the Rust type to an LLVM type and get the string representation of the LLVM type. Then append that to the intrinsic name.

That's how it worked before, we must define all those intrinsics in `context` though which is the annoying part.

Reply:
oh right :/ yea. there's no path around that unless we start registering intrinsics lazily or sth. But that's out of scope for this PR

Reply:
Looks like taking `ptr` works great :D

Reply:
yea. I presume LLVM thinks it's a pointer to a local variable and thus not something constant.